### DISCUSSION TITLE
DNSCrypt-Proxy Service - exists in UDM 3.X, but needs configuration

### DISCUSSION THREAD

**Discussion by ghost** [unifi-utilities/unifi-common#526] [Ideas]
UDM firmware 3.X comes with DNSCrypt-Proxy service that is already running, but isn't configured and GUI doesn't have any settings for it. I think UI will eventually include such GUI settings, but DNSCrypt-Proxy service should already be configurable through CLI. Has anyone configured it already? Official manual isn't very clear.

**Comment by boostchicken** [comment]
Did you ever find amything out about this?

**Comment by ghost** [comment]
Sadly I wasn't able to get it working. My knowledge of Linux is limited to basics. I was hoping someone would've figured it out by now.

**Comment by SamErde** [comment]
I haven't played with UDM 3.x firmware yet, but suspect they would have DNSCrypt-Proxy running on its own port and you'd need to point the DNS client at that port in order to use it. Curious if Ubiquiti have any plans for expanding how DNSCrypt-Proxy is used for the internals and/or for user-managed configuration within the platform. Beyond that piece of plumbing, it's not terribly hard to configure itself. There are many Pi-hole walkthroughs that include examples of configuring DNSCrypt-Proxy.
